src/coreclr/vm/exceptionhandling.cpp

```diff
@@ -7159,5 +7431,842 @@ void ExceptionTracker::ResetThreadAbortStatus(PTR_Thread pThread, CrawlFrame *pC
+        if (pExInfo->m_hThrowable)
+        {
+            if (!CLRException::IsPreallocatedExceptionHandle(pExInfo->m_hThrowable))
+            {
+                DestroyHandle(pExInfo->m_hThrowable);
+            }
+            pExInfo->m_hThrowable = NULL;
+        }
+        pExInfo->m_stackTraceInfo.FreeStackTrace();
+        pExInfo = pExInfo->m_pPrevExInfo;
+    }
+
+    pThread->GetExceptionState()->SetCurrentExInfo(pExInfo);
+
+    pThread->SafeSetLastThrownObject(NULL);
+
+    ExceptionTracker::UpdateNonvolatileRegisters(pvRegDisplay->pCurrentContext, pvRegDisplay, FALSE);
+    if (pHandlerIP != NULL)
+    {
+        ClrRestoreNonvolatileContext(pvRegDisplay->pCurrentContext);
+    }
+    else
+    {
+#ifdef HOST_UNIX
+        if (propagateExceptionCallback)
+        {
+            // A propagation callback was supplied.
+            STRESS_LOG3(LF_EH, LL_INFO100, "Deferring exception propagation to Callback = %p, IP = %p, SP = %p \n", propagateExceptionCallback, GetIP(pvRegDisplay->pCurrentContext), GetSP(pvRegDisplay->pCurrentContext));
+
+            UpdateContextForPropagationCallback(propagateExceptionCallback, propagateExceptionContext, pvRegDisplay->pCurrentContext);
+            GCX_PREEMP_NO_DTOR();
+            ClrRestoreNonvolatileContext(pvRegDisplay->pCurrentContext);
+        }
+#endif // HOST_UNIX
+        // Throw exception from the caller context
+#if defined(HOST_AMD64)
+        ULONG64* returnAddress = (ULONG64*)targetSp;
+        *returnAddress = pvRegDisplay->pCurrentContext->Rip;
+#elif defined(HOST_X86)
+        ULONG32* returnAddress = (ULONG32*)targetSp;
+        *returnAddress = pvRegDisplay->pCurrentContext->Eip;
+#elif defined(HOST_ARM64)
+        pvRegDisplay->pCurrentContext->Lr = GetIP(pvRegDisplay->pCurrentContext);
+#elif defined(HOST_ARM)
+        pvRegDisplay->pCurrentContext->Lr = GetIP(pvRegDisplay->pCurrentContext);
+#elif defined(HOST_RISCV) || defined(HOST_LOONGARCH64)
+        pvRegDisplay->pCurrentContext->Ra = GetIP(pvRegDisplay->pCurrentContext);
+#endif
+        // The exception needs to be stored in a place that is not going to be unwound, so
+        // use thread local storage
+        thread_local OBJECTREF oref;
+        oref = exceptionObj.Get();
+        SetIP(pvRegDisplay->pCurrentContext, (ULONG64)(void (*)(OBJECTREF))RealCOMPlusThrow);
+#if defined(HOST_AMD64)
+        SetSP(pvRegDisplay->pCurrentContext, targetSp - 8);
+#elif defined(HOST_X86)
+        SetSP(pvRegDisplay->pCurrentContext, targetSp - 4);
+#endif            
+#ifdef HOST_AMD64
+#ifdef UNIX_AMD64_ABI
+#define FIRST_ARG_REG Rdi
+#else
+#define FIRST_ARG_REG Rcx
+#endif
+#elif defined(HOST_X86)
+#define FIRST_ARG_REG Ecx
+#elif defined(HOST_ARM64)
+#define FIRST_ARG_REG X0
+#elif defined(HOST_ARM)
+#define FIRST_ARG_REG R0
+#elif defined(HOST_RISCV64) || defined(HOST_LOONGARCH64)
+#define FIRST_ARG_REG A0
+#endif
+
+#ifdef USE_CHECKED_OBJECTREFS
+        pvRegDisplay->pCurrentContext->FIRST_ARG_REG = (size_t)&oref;
```

Comment:
This makes an assumptions about calling convention used to pass `OBJECTREF`. I do not think that this will work on all platforms.

Would it be better to have a dedicated entrypoint that takes `Object*` and avoid the avoid the thread local, etc.?